### ISSUE TITLE
FR: Allow preventing elements from being hidden by marking them with a data attribute

### ISSUE THREAD

**Issue by diondiondion** [theKashey/aria-hidden#1]
Hi,

first of all, thanks so much for this library, it's super useful and reliably does exactly what it says on the tin. :)

We're using `aria-hidden` as part of our modal component, and it's working great. However, there's a conflict with our alert/toast elements that pop up to confirm actions.

All of our modals are rendered at the root of our app. The alerts are _also_ rendered at the root of our app, into a live region that makes sure that any changes inside of it are being announced to screen readers.

Now the conflict lies in the fact that when a modal is open, the live region used for alerts is marked as `aria-hidden="true"`, which effectively disables it. We can't dynamically move the alert container inside of the modal, and using the second parameter to limit `aria-hidden` to a common parent also doesn't work in our case.

Instead, we need a way to exclude a specific element from being hidden.

It'd be great if you'd consider defining a custom "ignore" attribute that we could add to elements that we don't want to be hidden by the library. (For example `data-aria-hidden-ignore`.)

Thank you!
Dion

**Comment by theKashey** [comment]
Absolutely valid use case.
Actually you can do it right now - `hideOthers` accepts an array as first element, and it will "skip" all nodes inside it

```js
function hideForModal(modalDomNode) {
  const otherNodes = [...document.querySelectorAll('[data-aria-hidden-ignore]')]
  // hide everything except Modal and all "otherNodes" 
  hideOthers([modalDomNode, ...otherNodes]);
}
```

**Comment by diondiondion** [comment]
> Actually you can do it right now - `hideOthers` accepts an array as first element, and it will "skip" all nodes inside it

Wow, thanks! I didn't realise that was possible. 👍  Would be worth adding to the docs.

Feel free to close this ticket. :)

**Comment by theKashey** [comment]
Your are welcome, and that's for the advice :)
